delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala

```diff
@@ -12,8 +11,8 @@ import ch.epfl.bluebrain.nexus.delta.sdk.model.{Label, ResourceRef}
 sealed trait SearchParams {
   def deprecated: Option[Boolean]
   def rev: Option[Long]
-  def createdBy: Option[Subject]
-  def updatedBy: Option[Subject]
+  def createdBy: Option[Iri]
```

Comment:
It is better to have `Subject` here, since in other parts of the code is better to deal with `Subject` than with `Iri`

Reply:
Yes but the problem is using a subject brings complexity during deserialization

Reply:
This issue in deserialization was already dealt with when implementing `Permissions`. Having an implicit BaseUri you can always extract the Iri from the Subject

Reply:
In Permissions, it must be another usage, here we extract the Iri from a get parameter to build the `SearchParams`.
To build a subject, we either have to add a discriminator in the get parameter or parse first as an Iri and then parse the Iri as a subject.

Reply:
I guess the directive which extracts the SearchParams should do that, right? And yes, we have to parse the Iri and from it generate an Identity.

In the Identity class we need an apply method as follows:
```scala
def apply(iri: Iri): Either[TokenRejection, Identity] = ???
```

Reply:
Is this refinement worth the amount of additional code and tests though ?

Reply:
I think we are going to need the conversion from `Iri => Identity` anyways (like we now have the one from Identity => Iri). So I would say yes.

But let's see what the others say.